util/confutil/metadata.go

```diff
@@ -0,0 +1,30 @@
+package confutil
+
+import (
+	"os"
+)
+
+// MetadataProvenanceMode is the type for setting provenance in the metdata file
+type MetadataProvenanceMode int
+
+const (
+	// MetadataProvenanceModeMin sets minimal provenance (default)
+	MetadataProvenanceModeMin MetadataProvenanceMode = iota
+	// MetadataProvenanceModeMax sets full provenance
+	MetadataProvenanceModeMax
+	// MetadataProvenanceModeOff doesn't set provenance
+	MetadataProvenanceModeOff
+)
+
+// MetadataProvenance returns the provenance mode to set in the metadata file
+func MetadataProvenance() MetadataProvenanceMode {
+	switch os.Getenv("BUILDX_METADATA_PROVENANCE") {
+	case "min":
+		return MetadataProvenanceModeMin
+	case "max":
+		return MetadataProvenanceModeMax
+	case "off":
```

Comment:
Do we have some precedence for "off" ? We do use "disabled" in `--attest`.

Additionally I think we should run it through `ParseBool` so that `=0` and `=false` work as well.